openslides_backend/action/agenda_item/update.py

```diff
@@ -22,3 +26,67 @@ class AgendaItemUpdate(UpdateAction):
             "duration",
         ]
     )
+
+    def calc_is_internal(self, type_: int, parent_is_internal: Optional[bool]) -> bool:
+        return type_ == AgendaItem.INTERNAL_ITEM or bool(parent_is_internal)
+
+    def calc_is_hidden(self, type_: int, parent_is_hidden: Optional[bool]) -> bool:
+        return type_ == AgendaItem.HIDDEN_ITEM or bool(parent_is_hidden)
+
+    def handle_children(
+        self, id_: int, parent_is_hidden: bool, parent_is_internal: bool
+    ) -> ActionPayload:
+        instances = []
+        agenda_item = self.datastore.get(
+            FullQualifiedId(self.model.collection, id_), ["child_ids"]
+        )
+        if agenda_item.get("child_ids"):
+            for child_id in agenda_item["child_ids"]:
+                child_ai = self.datastore.get(
+                    FullQualifiedId(self.model.collection, child_id), ["type"]
+                )
```

Comment:
Please use `get_many` here to save datastore calls. Just search for `GetManyRequest` in the actions

Reply:
Done.